### DISCUSSION TITLE
if workflow starting state's type is event, will 2 events trigger this state to run 2 times?

### DISCUSSION THREAD

**Discussion by elim19** [serverlessworkflow/specification#751] [Q&A] [answered]
for example, the following SW, if send OrderCreatedEvent twice, will there be two subFlowA running?
because if it is not starting state, there will be only one subFlowA running, why is starting state so special?

![image](https://user-images.githubusercontent.com/123845880/215436888-716f11f9-cad5-4ca8-8216-f8b21132d263.png)

**Comment by lsytj0413** [comment, marked as answer]
Aha, it depend on runtime implement, I think https://docs.camunda.io/docs/0.25/components/zeebe/bpmn-workflows/message-events/#message-start-events this will help you.

**Reply by elim19** [reply]
> https://docs.camunda.io/docs/0.25/components/zeebe/bpmn-workflows/message-events/#message-start-events

@lsytj0413 I check the article, so if the workflow is deployed and correlationKey is empty , it will trigger the event everytime there is a new event coming, right?

**Reply by elim19** [reply]
by the way, is camunda 8 an implementation of severless workflow?

**Reply by lsytj0413** [reply]
> > https://docs.camunda.io/docs/0.25/components/zeebe/bpmn-workflows/message-events/#message-start-events
> 
> @lsytj0413 I check the article, so if the workflow is deployed and correlationKey is empty , it will trigger the event everytime there is a new event coming, right?

I think so.

**Reply by lsytj0413** [reply]
> by the way, is camunda 8 an implementation of severless workflow?

No, camunda 8 is base on zeebe workflow engine, it's an implementation of BPMN2.0

**Reply by elim19** [reply]
> > by the way, is camunda 8 an implementation of severless workflow?
> 
> No, camunda 8 is base on zeebe workflow engine, it's an implementation of BPMN2.0

@lsytj0413 , I am confused, when you mention "it depend on runtime implement", why are you referencing "camunda 8"?

**Reply by tsurdilo** [reply]
i think he was just replying to your question https://github.com/serverlessworkflow/specification/discussions/751#discussioncomment-4817502
which included that version. No camunda release versions that i am aware of use SW and I am not aware of any future integration work (noone reached out), if that helps.

**Reply by lsytj0413** [reply]
> > > by the way, is camunda 8 an implementation of severless workflow?
> > 
> > 
> > No, camunda 8 is base on zeebe workflow engine, it's an implementation of BPMN2.0
> 
> @lsytj0413 , I am confused, when you mention "it depend on runtime implement", why are you referencing "camunda 8"?

Referencing `camunda 8` as an example, BPMN2.0 also has an element Message-start-event （same as swf's starting event），If the user didn't specified `correlationKey`, multi event will trigger multi ProcessInstance （Workflow Instance）
